src/useImageDimensions.ts

```diff
@@ -0,0 +1,57 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+class Dimensions {
```

Comment:
To me it seems a bit unnecessary to introduce en entirely new class just to avoid dividing two numbers. Don't you agree?

What do you think about just calculating the `aspectRation` when you call `setState`?

Reply:
Please, make remarks only with references to generally accepted rules and patterns. My position is: "DRY". I will leave this class as it is.

Reply:
How is having a class with a gett DRY? I fail to see how you are repeating something with my suggestion 🤔 

Avoiding over-engineering is, in my opinion, a very generally accepted rule. It's stated as one of the main point in Google's guide to code review: https://google.github.io/eng-practices/review/

Reply:
I'm tired of this discussion, really. Where had you seen here an over-engineering? Everybody knows what is classes.